File: server/models/pokemon.py
from typing import List

from pydantic import BaseModel


class Pokemon(BaseModel):
    pokemon_id: str
    number: int
    name: str
    type_one: str
    type_two: str
    total: str
    hit_points: int
    attack: int
    defense: int
    special_attack: int
    special_defense: int
    speed: int
    generation: int
    legendary: bool
    icon_url: str
    is_captured: bool = False

File: client/src/Models/Pokemon.ts
export interface Pokemon {
    pokemon_id: number
    number: number
    name: string
    type_one: string
    type_two: string
    total: string
    hit_points: number
    attack: number
    defense: number
    special_attack: number
    special_defense: number
    speed: number
    generation: number
    legendary: boolean
    icon_url: string
    is_captured: boolean
}